src/inmanta/agent/agent.py

```diff
@@ -800,14 +810,17 @@ async def get_facts(self, resource: JsonType) -> Apireturn:
             return await executor.get_facts(resource_refs[0])
 
     async def setup_executor(
-        self, version: int, action: const.ResourceAction, resources: list[JsonType]
+        self, version: int, action: const.ResourceAction, resources: list[JsonType], resource_types: set[ResourceType]
     ) -> tuple[dict[ResourceVersionIdStr, const.ResourceState], list[ResourceDetails], Optional[executor.Executor]]:
-        # TODO refactor wiring of tuples around -> not sure how
-
         """
         Setup an executor for resource deployment
 
-        returns a tuple
+        :param version: model version being deployed
+        :param action: the resource action associated with this deploy
+        :param resources: list of resources in serialized form: a dict with information about
```

Comment:
As far as I can tell we now no longer need `resources` to actually set up the executor. Perhaps we can extract the executor setup from this method? i.e. give this one a different name (because it does a lot more than set up the executor, and have it call into the new `setup_executor(version, resource_types) -> Optional[executor.Executor])`?